dandischema/datacite/tests/test_datacite.py

```diff
@@ -35,21 +37,178 @@ def datacite_post(datacite: dict, doi: str) -> None:
     rg = requests.get(url=f"https://api.test.datacite.org/dois/{doi}/activities")
     rg.raise_for_status()
 
-    # cleaning url
-    _clean_doi(doi)
+    if clean:
+        # cleaning url
+        _clean_doi(doi)
+
+
+def datacite_update(datacite: dict, doi: str) -> None:
+    """Update the datacite object and check the status of the request"""
+    rp = requests.put(
+        url=f"https://api.test.datacite.org/dois/{doi}",
+        json=datacite,
+        headers={"Content-Type": "application/vnd.api+json"},
+        auth=("DARTLIB.DANDI", os.environ["DATACITE_DEV_PASSWORD"]),
+    )
+    rp.raise_for_status()
+
+    # checking if i'm able to get the url
+    rg = requests.get(url=f"https://api.test.datacite.org/dois/{doi}/activities")
+    rg.raise_for_status()
 
 
 def _clean_doi(doi: str) -> None:
     """Remove doi. Status code is ignored"""
-    requests.delete(
+    rq = requests.delete(
         f"https://api.test.datacite.org/dois/{doi}",
         auth=("DARTLIB.DANDI", os.environ["DATACITE_DEV_PASSWORD"]),
     )
+    print("\n in _clean_doi", doi, rq.status_code)
+    return rq.status_code
+
+
+@pytest.mark.skip(
+    reason="to not produced too many dois, not sure if we want to keep it as a test"
+)
+def test_datacite_lifecycle() -> None:
+    """testing the lifecycle of a public dandiset and doi (from draft to published)"""
+
+    # checking which doi is available
+    doi_available = False
+    while not doi_available:
+        dandi_id = f"000{random.randrange(500, 999)}"
+        print(f"searching for available doi, trying dandi_id: {dandi_id}")
+        doi_root = f"10.80507/dandi.{dandi_id}"
+        if _clean_doi(doi_root) != 405:
+            doi_available = True
+            print(f"found available doi, dandi_id: {dandi_id}")
+
+    dandi_id_prefix = f"DANDI:{dandi_id}"
+    # creating the main/root doi and url
+    doi_root = f"10.80507/dandi.{dandi_id}"
+    url_root = f"https://dandiarchive.org/dandiset/{dandi_id}"
+
+    # creating draft dandiset with minimal metadata
+    version = "draft"
+    meta_dict = {
+        "identifier": dandi_id_prefix,
+        "id": f"{dandi_id_prefix}/{version}",
+        "name": "Testing Dataset: lifecycle",
+        "description": "testing lifecycle of a dataset and doi: draft",
+        "version": version,
+        "contributor": [
+            {
+                "name": "A_last, A_first",
+                "email": "nemo@example.com",
+                "roleName": [RoleType("dcite:ContactPerson")],
+                "schemaKey": "Person",
+            }
+        ],
+        "license": [LicenseType("spdx:CC-BY-4.0")],
+        "citation": "A_last, A_first 2021",
+        "manifestLocation": [
+            f"https://api.dandiarchive.org/api/dandisets/{dandi_id}/versions/{version}/assets/"
+        ],
+        "assetsSummary": {
+            "schemaKey": "AssetsSummary",
+            "numberOfBytes": 10,
+            "numberOfFiles": 1,
+        },
```

Comment:
hopefully does not matter but whenever dandiset is created, there is no assets summary yet -- it is done async IIRC, hence

```suggestion
```

Reply:
`assetsSummary` is a mandatory field for `Dandiset`

Reply:
in dandi archive we do following if not provided in metadata

```
        if 'assetsSummary' not in metadata:
            metadata['assetsSummary'] = {
                'schemaKey': 'AssetsSummary',
                'numberOfBytes': 0,
                'numberOfFiles': 0,
            }
```

http://github.com/dandi/dandi-archive/blob/HEAD/dandiapi/api/models/version.py